reframe/core/schedulers/__init__.py

```diff
@@ -90,6 +90,15 @@ def finished(self, job):
         :meta private:
         '''
 
+    @abc.abstractmethod
+    def poll_jobs(self, jobs):
+        '''Poll all the requested jobs.
+
+        :arg jobs: A list of job descriptors.
+
+        :meta private:
+        '''
+
```

Comment:
> The poll_jobs() above should not be abstract, but it should provide the naive implementation, which will go over the jobs list and call poll() on each of the jobs.

The naive implementation will be in the abstract class `JobScheduler`? In cases like slurm where we have the custom aggregate `poll_jobs()` will we have the following?
```
def poll_job(self, job):
    self.poll_jobs([job])
```
Otherwise we will have to have a lot of duplicate code. I made the aggregate `poll_jobs` an abstract method because this is the one that will be called from the policies. I think `poll_job` will be only used from unittests

Reply:
Why don't we define it as `def poll(*jobs)` and have the best of both worlds? You would be able to call it either with a single job or with a list of jobs as `poll(single_job)` or `poll(*multiple_jobs)`.